### ISSUE TITLE
Dart API docs for analyzer Package are missing

### ISSUE THREAD

**Issue by SebastianWendorf** [dart-lang/sdk#55936]
The Dart API docs page for the analyzer package seems to be missing: https://pub.dev/documentation/analyzer/latest/

Furthermore there seem to be some analysis issues with the package in general: https://pub.dev/packages/analyzer/score

Is there some alternative place where I can find the documentation?

**Comment by keertip** [comment]
@jonasfj , @athomas , any idea what might be happening here? Are the docs not getting generated?

**Comment by jonasfj** [comment]
See:
https://pub.dev/packages/analyzer/score

This is probably a macro issue...
Since package:macro depends on _macro package from the SDK, it doesn't work
with stable SDK.
But since it's an indirect dependency we can detect SDK incompatibility.
Fix could be for analyzer to make SDK constraint more explicit. Or for
macros to be an actual SDK package, which we could then handle in Pana --
not that said infrastructure is in place yet.

**Comment by bwilkerson** [comment]
@scheglov

**Comment by scheglov** [comment]
@jakemac53

**Comment by jakemac53** [comment]
I filed an issue related to this somewhere.... there isn't any reasonable solution that has been proposed afaik... this version of analyzer can only be analyzed on the absolute latest SDK.

**Comment by jakemac53** [comment]
Ah, https://github.com/dart-lang/pub-dev/issues/7597 is the core issue

**Comment by jonasfj** [comment]
Add a note to:
https://github.com/dart-lang/pub-dev/issues/7597#issuecomment-2151065865

I think making package:macros an SDK would make the SDK constraints more explicit. And this would enable tools to be smarter.

This might be less painful when the latest version works with the latest stable SDK release, but whenever this is not the case packages using macros will feel more pain.

**Comment by jonasfj** [comment]
Hmm, I guess if we really want to pretend `package:macros` isn't an SDK package, we could hardcode into the SDK which versions its compatible with, and make all tooling understand that 🤣 -- but that was sort of what SDK packages was for in the first place (I think 🙈).

**Comment by srawlins** [comment]
We still have no latest docs (I suspect we'll get some when Dart 3.5.0 is released).

@jakemac53 is there movement to change how analyzer depends on macros or macros depends on _macros, such that we won't always be needing to publish an analyzer that can _only_ work on the latest dev builds of the Dart SDK?

**Comment by jakemac53** [comment]
Yes, we are trying to resolve the issue, but it will take a while
